### DISCUSSION TITLE
提供一个解析 Ethereum ABI 并展示的组件

### DISCUSSION THREAD

**Discussion by jeasonstudio** [ant-design/ant-design-web3#542] [Ideas]
许多应用都有展示 ABI 为表单或详情的需求（大部分为基础设施应用，比如浏览器、IDE、合约调试应用、交易分析应用等等），我们可以尝试在 AntDesign Web3 中提供出来；

以太坊 abi 的规范是一个类似 JSON Schame 的数组结构，可以考虑用类似 schema form 的方式解析为表单或详情：

规范：https://docs.soliditylang.org/en/v0.8.23/abi-spec.html
一些例子：
<img width="580" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/17971291/521239fb-bb5f-4ac5-b35f-7118b8f77a33">
<img width="307" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/17971291/00ad124c-f100-4714-abd1-13d8b28b7ee9">
<img width="924" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/17971291/72e29c2a-19e4-4ecd-a0a5-aea19815968a">

**Comment by Likang0122** [comment]
上传abi然后我们解析成这样吗

**Comment by yutingzhao1991** [comment]
确实，这个组件我也写过，在 IDE 里面

**Comment by yutingzhao1991** [comment]
具体怎么定位，组件上如何抽象，是否要考虑其它链，细节 @jeasonstudio 看看能不能想一想
